modules/compute-api/src/main/java/org/apache/ignite/compute/ComputeJob.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compute;
+
+/**
+ * A Compute job that may be executed on an Ignite node (or a few nodes, or on the whole cluster).
+ *
+ * @param <R> job result type
+ */
+public interface ComputeJob<R> {
+    /**
+     * Executes the job on an Ignite node.
+     *
+     * @param context  context
+     * @param args     job arguments
```

Comment:
This javadoc will have to be expanded in the future, right? Currently, I don't understand supported argument types, for example

Reply:
Yes, it should be extended in the future. Moreover, currently, there is only the execution method, so no ability to split one task to few jobs. Maybe I'm wrong, but I think this will be changed on next phases, so right now it doesn't make a lot of sense to write a lot in the Javadoc.

As for the supported types, this will also change. On early phases we will only support simple argument types (approximately, primitives, strings, byte arrays, maybe something else), but this restriction will probably be lifted (or weakened) in the future. So, for now we cannot tell a lot in the Javadoc on this matter.